openmetadata-service/src/main/java/org/openmetadata/service/jdbi3/AlertRepository.java

```diff
@@ -0,0 +1,269 @@
+    daoCollection
+        .entityExtensionTimeSeriesDao()
+        .insert(alertId.toString(), alertActionId.toString(), "alertActionStatus", JsonUtils.pojoToJson(currentStatus));
+    return currentStatus;
+  }
+
+  public void setStatus(UUID alertId, UUID alertActionId, AlertActionStatus status) throws IOException {
+    daoCollection
+        .entityExtensionTimeSeriesDao()
+        .insert(alertId.toString(), alertActionId.toString(), "alertActionStatus", JsonUtils.pojoToJson(status));
+  }
+
+  @SneakyThrows
+  public void updateAlertActionPublishers(Alert alert) {
+    // Delete existing alert action publisher and create with the updated Alert
+    Alert storedAlert = get(null, alert.getId(), getFields("*"));
+    if (alert.getAlertActions().equals(storedAlert.getAlertActions())) {
+      EntityRepository<AlertAction> alertActionEntityRepository = Entity.getEntityRepository(ALERT_ACTION);
+      for (EntityReference alertActionRef : alert.getAlertActions()) {
+        AlertAction action =
+            alertActionEntityRepository.get(null, alertActionRef.getId(), alertActionEntityRepository.getFields("*"));
+        if (Boolean.TRUE.equals(action.getEnabled())) {
+          // If there was a previous alert action either in disabled state or stopped due
+          // to errors, update it and restart publishing
+          AlertsActionPublisher previousPublisher = getAlertActionPublisher(alert.getId(), action);
+          if (previousPublisher == null) {
+            addAlertActionPublisher(alert, action);
+            return;
+          }
+
+          // Update the existing publisher
+          AlertActionStatus.Status status = previousPublisher.getAlertAction().getStatusDetails().getStatus();
+          previousPublisher.updateAlertWithAlertAction(alert, action);
+          if (status != AlertActionStatus.Status.ACTIVE && status != AlertActionStatus.Status.AWAITING_RETRY) {
+            // Restart the previously stopped publisher (in states notStarted, error, retryLimitReached)
+            BatchEventProcessor<EventPubSub.ChangeEventHolder> processor =
+                EventPubSub.addEventHandler(previousPublisher);
+            previousPublisher.setProcessor(processor);
+            LOG.info("Webhook publisher restarted for {}", alert.getName());
+          }
+        } else {
+          deleteAlertActionPublisher(alert.getId(), action);
+        }
+      }
+    } else {
+      // if some alerts are removed or added
+      deleteAlertAllPublishers(alert.getId());
+      addAlertActionPublishers(alert);
+    }
+  }
+
+  public void deleteAlertActionPublisher(UUID alertId, AlertAction action) throws InterruptedException {
+    List<AlertsActionPublisher> alertPublishers = alertPublisherMap.get(alertId);
+    int position = -1;
+    for (int i = 0; i < alertPublishers.size(); i++) {
+      AlertsActionPublisher alertsActionPublisher = alertPublishers.get(i);
+      if (alertsActionPublisher.getAlertAction().getId().equals(action.getId())) {
```

Comment:
This is bit confusing, it looks like you are trying to find the alertPubliser from map and you are looping through alertActionPublisher to find and delete it?